esp32/moduhashlib.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the Micro Python project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014 Paul Sokolovsky
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#include "py/runtime.h"
+
+#include "mbedtls/sha256.h"
+
+#if MICROPY_PY_UHASHLIB_SHA1
+#include "mbedtls/sha1.h"
+#endif
+
+union sha_ctxs {
+        mbedtls_sha256_context sha256;
+#if MICROPY_PY_UHASHLIB_SHA1
+        mbedtls_sha1_context sha1;
+#endif
+};
+typedef struct _mp_obj_hash_t {
+    mp_obj_base_t base;
+    union sha_ctxs state;
+} mp_obj_hash_t;
+
+STATIC mp_obj_t sha256_update(mp_obj_t self_in, mp_obj_t arg);
+#if MICROPY_PY_UHASHLIB_SHA1
```

Comment:
It doesn't look like SHA1 is enabled, because it's still disabled in mpconfigport.h.  Since SHA1 is always available from the mbedtls code I think it's simpler to just have it unconditionally enabled in this file, ie remove all the #if's for UHASHLIB_SHA1.